### DISCUSSION TITLE
Search and Edit Files?

### DISCUSSION THREAD

**Discussion by peterbenjamin-kh** [charmbracelet/soft-serve#520] [Ideas]
While using `soft browse` to browse a repository, I find myself hitting `/` to fuzzy search in **Files** tab and hitting `e` on a file to edit it using my `$EDITOR`.

**Comment by aymanbagabas** [comment]
This is a great idea! If you're interested in working on this feature, please feel free to send a PR 🙂

**Reply by peterbenjamin-kh** [reply]
I can take a stab at it.

For fuzzy searching, I am thinking of using [`gum filter`](https://github.com/charmbracelet/gum/blob/main/filter/filter.go) as an inspiration, unless you have a better reference.

**Reply by aymanbagabas** [reply]
Soft Serve uses the [list bubble](https://github.com/charmbracelet/bubbles/blob/master/list/list.go), maybe that would be a better start.
https://github.com/charmbracelet/soft-serve/blob/main/pkg/ui/pages/repo/files.go#L81
